arm-iothub/2016-02-03/swagger/iothub.json

```diff
@@ -1208,18 +1236,18 @@
       }
     },
     "CloudToDeviceProperties": {
-      "description": "The Iot Hub Cloud-To-Device messaging properties.",
+      "description": "The IoT hub cloud-to-device messaging properties.",
       "type": "object",
       "properties": {
         "maxDeliveryCount": {
-          "description": "The max delivery count for the device queue. Range : 1-100.",
+          "description": "The max delivery count for cloud-to-device messages in the device queue. See: https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-messaging#cloud-to-device-messages.",
```

Comment:
It makes sense to add defaults for some of the properties, however there are some properties for which the default value of that property is dependent on the value of another property. For instance the default value of the event hub partition count is dependent on the Sku that is selected. The range of values also have a similar constraint for some properties. I didn't find a way to express this in swagger. So the best approach to me really is to have a single documentation overwrite for all default values and range of values which express all such conditions appropriately.